exercises/practice/custom-set/src/test/java/CustomSetTest.java

```diff
@@ -187,24 +187,26 @@ public void addToEmptySet() {
 
         actual.add(element);
 
-        assertNotNull(actual);
-        assertFalse(actual.isEmpty());
-        assertTrue(expected.equals(actual));
+        assertThat(actual).isNotNull().returns(true, s -> s.equals(expected));
```

Comment:
I'm not sure this is an improvement over just using separate assertions:

```java
assertThat(actual).isNotNull();
assertThat(actual.isEmpty()).isFalse();
assertThat(actual.equals(expected)).isTrue();
```

The main reason for saying this is that students will have a look at the test code when some tests are failing to help them out, and they shouldn't have to spend a lot of extra effort understanding what the tests are doing. As far as test code goes, the simpler the better.

This comment addresses all uses of `assertThat(...).returns()` in the tests, to save on the amount of review comments in this PR :)

Reply:
I can do so but isn't the whole point of using AssertJ that it has a better API for doing this? Shouldn't putting each new test on a new line already address the issue you just raised?

I do agree, though, that the `returns` call is not necessarily the most beautiful code.

Reply:
Ah yes I didn't express myself clearly I see. I don't have anything against chaining AssertJ assertions per se. For instance, this is totally clear:

```java
assertThat(someList)
    .isNotNull()
    .containsExactly(1, 2, 3, 4);
```

My problem in this case is with the `.returns` method. The syntax is a bit cumbersome, and probably requires students to look up what it does, which makes the tests more complex than needed. The intention of my comment therefore was to replace the uses of `.returns` with separate assertions.

Reply:
Ah okay. Yes, I'll remove that then.